### DISCUSSION TITLE
Terraform and provider dependancies

### DISCUSSION THREAD

**Discussion by Gilles95** [renovatebot/renovate#8101] [Request Help]
Hello,

I see in the documentation, dependancy can be checked for terraform module. Should that also be the case for the terraform provider version? example below:

```
provider "aws" {
  region = "us-east-1"
  version = "1.0"
}
```

```
required_providers {
  mycloud = {
    source  = "mycorp/mycloud"
    version = "~> 1.0"
  }
```

Thanks

Gilles

**Comment by rarkins** [comment]
A quick search of the source seems to indicate yes: https://github.com/renovatebot/renovate/search?q=required_providers

**Comment by viceice** [comment]
@Gilles95 This should already work. Please use the latest terraform version, as putting the version to the provider config is deprecated and fails.

Here a working `versions.tf` sample:
```hcl
terraform {
  required_providers {
    docker = {
      source  = "kreuzwerker/docker"
      version = "2.8.0"
    }
    helm = {
      source  = "hashicorp/helm"
      version = "1.3.2"
    }
    kubernetes = {
      source  = "hashicorp/kubernetes"
      version = "1.13.3"
    }

    kubectl = {
      source  = "gavinbunney/kubectl"
      version = "1.9.4"
    }
  }
  required_version = "~> 0.14.0"
}

```

**Reply by viceice** [reply]
`required_version` is currently not handled by renovate, but a simple regex manager can be used:

```json
    {
      "fileMatch": ["\\.tf$"],
      "matchStrings": ["required_version\\s*=\\s*\"(?<currentValue>.+?)\""],
      "datasourceTemplate": "docker",
      "versioningTemplate": "hashicorp",
      "depNameTemplate": "hashicorp/terraform"
    }
```

**Reply by Gilles95** [reply]
Thanks a lot.

**Reply by ekhaydarov** [reply]
@viceice ive just copy pasted your suggestions and it does not seem to work
it looks like this. Any clues as to what im missing?
```{
             "packageFile": "aad/azureff/versions.tf",
             "deps": [
               {
                 "depName": "hashicorp/terraform",
                 "currentValue": ">= 0.15",
                 "datasource": "docker",
                 "versioning": "hashicorp",
                 "replaceString": "required_version = \">= 0.15\"",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "sourceUrl": "https://github.com/hashicorp/terraform",
                 "currentVersion": "1.2.6"
               }
             ],
             "matchStrings": ["required_version\\s*=\\s*\"(?<currentValue>.+?)\""],
             "depNameTemplate": "hashicorp/terraform",
             "datasourceTemplate": "docker",
             "versioningTemplate": "hashicorp"
           },
```

**Reply by secustor** [reply]
@ekhaydarov  the regex manager shouldn't be necessary anymore as the Terraform  supports it now natively

**Reply by ekhaydarov** [reply]
@secustor thanks for the heads up. However, I have tried a form of that too an am still unable to get it right. Would you be kind enough to point me in the right direction. I need to have a separate PR for updating only the `required_version` in versions.tf. I have followed what I have seen elsewhere and added this to package rules

```
  terraform: {
    fileMatch: ["(^|/)versions\\.tf$", "(^|/)terragrunt\\.hcl$"],
    ignorePaths: ["**/main.tf", "**/data.tf", "**/backend.tf", "**/providers.tf"]
  },
  packageRules: [
    {
      groupName: "Terraform required version",
      groupSlug: "terraform_required_version",
      matchPackageNames: ["hashicorp/terraform"]
    }
  ]
```

I am still coming up short. it seems to find the required version which looks like this `required_version = ">= 1.1.0"`. But the updated block is empty or it states that there is a config issue.

I'm missing something simple in the docs? Let me know if youd like me to open a separate issue, but it seems a search to lead here should be beneficial to future searches

Thanks

**Reply by secustor** [reply]
Can you create a reproduction repository?
